playback/src/player.rs

```diff
@@ -43,6 +44,52 @@ enum PlayerCommand {
     Seek(u32),
 }
 
+#[derive(Clone, Copy, Debug)]
+struct NormalisationData {
+    track_gain_db: f32,
+    track_peak: f32,
+    album_gain_db: f32,
+    album_peak: f32,
+}
+
+impl NormalisationData {
+    fn new<T: Read + Seek>(file: &mut AudioDecrypt<T>) -> NormalisationData {
```

Comment:
Function should be called `parse_from_file` or something, and it's argument can just be a `mut file: T`
`AudioDecrypt<T>` (and by extension `&mut AudioDecrypt<T>`) already directly implement `Read + Seek`

Reply:
Also it can return an `io::Result<NormalisationData>` and use `?` instead of `unwrap`, moving the unwrap to the caller.

Reply:
have changed it to return a ```Result```, but I'm using ```match``` instead of ```?``` as I personally don't like the ```?``` operator as it's confusing (feels like it should mean optional) and feel that match is more readable, also as normalisation is not something that should fail hard if the data can't be read, I've set the behaviour to ```warn``` if the normalisation data cannot be read, and set it to the default value of ```1.0```